packages/react-i18n-webpack-plugin/src/react-i18n-webpack-plugin.ts

```diff
@@ -0,0 +1,264 @@
+    virtualModules.apply(compiler);
+
+    compiler.hooks.compilation.tap(
+      PLUGIN_NAME,
+      (
+        compilation: webpack.compilation.Compilation,
+        {
+          normalModuleFactory,
+        }: {normalModuleFactory: webpack.compilation.NormalModuleFactory},
+      ) => {
+        const handler = (parser: any) => {
+          new FindI18nImportPlugin().apply(parser);
+
+          // replace useI18n & withI18n call arguments
+          parser.hooks.evaluate
+            .for('CallExpression')
+            .tap(PLUGIN_NAME, (originalExpression: CallExpression) => {
+              if (
+                parser.state.module.resource.indexOf('node_modules') !== -1 &&
+                !parser.state.module.resource.endsWith('tsx') &&
+                !parser.state.i18nImports
+              ) {
+                return;
+              }
+
+              const componentPath = parser.state.module.resource;
+              const componentDir = parser.state.module.context;
+              const identifierName = parser.state.i18nImports.get(
+                componentPath,
+              );
+
+              if (
+                !identifierName ||
+                originalExpression.callee.type !== 'Identifier'
+              ) {
+                return;
+              }
+
+              let expression: CallExpression | null = null;
+              if (originalExpression.callee.name === 'compose') {
+                const foundExpressions: Array<
+                  Expression | SpreadElement
+                > = originalExpression.arguments.filter(
+                  node =>
+                    node.type === 'CallExpression' &&
+                    node.callee.type === 'Identifier' &&
+                    node.callee.name === identifierName,
+                );
+
+                if (foundExpressions.length > 0) {
+                  expression = foundExpressions[0] as CallExpression;
+                }
+              } else if (originalExpression.callee.name === identifierName) {
+                expression = originalExpression;
+              }
+
+              // skip calls where consumer manually added arguments
+              if (!expression || expression.arguments.length > 0) {
+                return;
+              }
+
+              const translationFiles = getTranslationFiles(parser);
+              if (translationFiles.length === 0) {
+                return;
+              }
+
+              // check if fall back exist
+              const fallBackExist = translationFiles.find(
+                translationFile =>
+                  translationFile === `${this.options.fallbackLocale}.json`,
+              );
+
+              const fallBackFileRelativePath = path.join(
+                './',
+                TRANSLATION_DIRECTORY_NAME,
+                `${this.options.fallbackLocale}.json`,
+              );
+
+              if (!fallBackExist) {
+                compilation.errors.push(
```

Comment:
Do we need to error?

Reply:
It's up to us. IMO I would like to warm the consumer that the fallback does not exist and it should be fix.
I debate between error or warning but landed on error since this should NEVER happen (ie. a translation folder exist with .json files but no fallback)

Reply:
Would it be definitely wrong if they had no fallback? I don't think we know it will be, and it's possible it could be an explicit choice. In the end, we should just stick to whatever the current plugin does to cut some scope.